### DISCUSSION TITLE
test-bitcoincore-rpc: A very not offical crate for creating a test Bitcoin Core RPC server

### DISCUSSION THREAD

**Discussion by casey** [rust-bitcoin/rust-bitcoin#1305] [Other Bitcoin-related crates]
This is something that I've needed for testing projects that interact with Bitcoin Core via RPC. It provides a very simple interface for populating blocks and transactions, and then serves those blocks and transactions via the standard RPC API. If anyone's interested, take a look! This is a crate that I wrote and maintain (along with @raphjaph), and not an official rust-bitcoin crate, so keep that in mind. It's primarily for our own needs while working on [ord](https://github.com/casey/ord), but simple PRs that extend it are welcome. 

- [repo](https://github.com/casey/ord/tree/master/test-bitcoincore-rpc)
- [crates.io](https://crates.io/crates/test-bitcoincore-rpc)
- [docs](https://docs.rs/test-bitcoincore-rpc/latest/test_bitcoincore_rpc/)

**Comment by sanket1729** [comment]
Have you looked at `bitcoind` https://crates.io/crates/bitcoind? We use it in rust-miniscript CI. I think the bdk and electrs folks also use it in their CI. It internally uses https://docs.rs/bitcoincore-rpc/0.16.0/bitcoincore_rpc/trait.RpcApi.html that interacts with rust-bitcoin data structures.

**Reply by casey** [reply]
We used the `bitcoind` crate for a bit, as well as running against whatever `bitcoind` binary was installed on the host system, and found that it was much slower than the dummy RPC API. The binary we're working on has a very large number of tests, and running against a live `bitcoin` made the edit/test/debug loop quite slow.
